gui/fitpanelv7/src/RFitPanel6.cxx

```diff
@@ -0,0 +1,512 @@
+       vect1.emplace_back("1", "MIGRAD");
+       vect1.emplace_back("2", "SIMPLEX");
+       vect1.emplace_back("3", "SCAN");
+       vect1.emplace_back("4", "Combination");
+
+       // corresponds to library == 2
+       model.fMethodMinAll.emplace_back();
+       std::vector<ROOT::Experimental::RComboBoxItem> &vect2 = model.fMethodMinAll.back();
+       vect2.emplace_back("1", "FUMILI");
+
+       // corresponds to library == 3
+       model.fMethodMinAll.emplace_back();
+       // std::vector<ROOT::Experimental::RComboBoxItem> &vect3 = model.fMethodMinAll.back();
+       // vect3.emplace_back("1", "Lib3_1");
+       // vect3.emplace_back("2", "Lib3_2");
+
+       // corresponds to library == 4
+       model.fMethodMinAll.emplace_back();
+       std::vector<ROOT::Experimental::RComboBoxItem> &vect4 = model.fMethodMinAll.back();
+       vect4.emplace_back("1", "TMVA Genetic Algorithm");
+
+       // select items list for initial display
+       model.fMethodMin = model.fMethodMinAll[model.fLibrary];
+       // model.fTypeXY = model.fTypeXYAll[model.fTypeId];
+
+       if (fHist) {
+          model.fMinRange = fHist->GetXaxis()->GetXmin();
+          model.fMaxRange = fHist->GetXaxis()->GetXmax();
+
+          model.fUpdateMinRange = fHist->GetXaxis()->GetXmin();
+          model.fUpdateMaxRange = fHist->GetXaxis()->GetXmax();
+       }
+
+       //defined values
+       model.fStep = (model.fMaxRange - model.fMinRange) / 100;
+       model.fRange[0] = model.fMinRange;
+       model.fRange[1] = model.fMaxRange;
+
+       model.fUpdateRange[0] = model.fUpdateMinRange;
+       model.fUpdateRange[1] = model.fUpdateMaxRange;
+       //model.fOperation = 0;
+       model.fFitOptions = 3;
+       model.fRobust = false;
+       model.fLibrary = 0;
+       model.fPrint = 0;
+
+
+       //Checkboxes Values
+       model.fIntegral = false;
+       model.fWeights = false;
+       model.fBins = false;
+       //model.fUseRange = false;
+       model.fAddList = false;
+       model.fUseGradient = false;
+       model.fSame = false;
+       model.fNoStore = false;
+       model.fMinusErrors = false;
+       //model.fImproveFit = false;
+
+       if(model.fNoStore){
+          model.fNoDrawing = true;
+       }
+       else{
+          model.fNoDrawing = false;
+       }
+
+       if((model.fFuncChangeInt >= 6) && (model.fFuncChangeInt <= 15)){
+          model.fLinear = true;
+
+       }
+       else {
+          model.fLinear = false;
+
+       }
+
+       //Communication with the JSONModel in JS
+       // TString json = TBufferJSON::ConvertToJSON(&model, gROOT->GetClass("FitPanelModel"));
+       TString json = TBufferJSON::ToJSON(&model);
+
+       fWindow->Send(fConnId, std::string("MODEL:") + json.Data());
+
+       return;
+   }
+
+   if (arg == "CONN_CLOSED") {
+      printf("FitPanel connection closed\n");
+      fConnId = 0;
+      return;
+   }
+
+   if (arg.find("DOFIT:") == 0) {
```

Comment:
Please use `arg.compare(0, 6, "DOFIT:")`, also adjust below. Else the whole string is searched. In general, it might be nicer to refactor:
```
auto posColumn = arg.find(':');
if (posColumn == std::string::npos) {
...
   return;
}
// else:
std::string_view command = arg.substr(0, posColumn);
std::string_view commandArg = arg.substr(posColumn + 1);
if (command == "DOFIT") {
   DoFit(commandArg);
} else if (...
```